nsxt/resource_nsxt_edge_cluster.go

```diff
@@ -0,0 +1,309 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+func resourceNsxtEdgeCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtEdgeClusterCreate,
+		Read:   resourceNsxtEdgeClusterRead,
+		Update: resourceNsxtEdgeClusterUpdate,
+		Delete: resourceNsxtEdgeClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"revision":     getRevisionSchema(),
+			"description":  getDescriptionSchema(),
+			"display_name": getDisplayNameSchema(),
+			"tag":          getTagsSchema(),
+			"edge_ha_profile_id": {
+				Type:        schema.TypeList,
+				Description: "Edge high availability cluster profile Ids",
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"bridge_ha_profile_id": {
+				Type:        schema.TypeList,
+				Description: "Bridge high availability cluster profile Ids",
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"member_node_type": {
+				Type:        schema.TypeString,
+				Description: "Node type of the cluster members",
+				Computed:    true,
+			},
+			"member": {
+				Type:        schema.TypeList,
+				Description: "Edge cluster members",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"description": {
+							Type:        schema.TypeString,
+							Description: "Description of this resource",
+							Optional:    true,
+						},
+						"display_name": {
+							Type:        schema.TypeString,
+							Description: "The display name of this resource. Defaults to ID if not set",
+							Optional:    true,
+							Computed:    true,
+						},
+						"member_index": {
+							Type:        schema.TypeInt,
+							Description: "System generated index for cluster member",
+							Computed:    true,
+						},
+						"transport_node_id": {
+							Type:        schema.TypeString,
+							Description: "UUID of edge transport node",
+							Required:    true,
+						},
+					},
+				},
+			},
+			"node_rtep_ips": {
```

Comment:
According to the spec, this whole section should be Computed, is that correct?

Reply:
Indeed. I'm not sure whether we need it or not too. What do you think?

Reply:
I think those IPs could be used elsewhere in config, so lets leave them